### ISSUE TITLE
DBD::Pg 3.16.0 tests fail

### ISSUE THREAD

**Issue by depesz** [bucardo/dbdpg#107]
I tried to install dbd::pg on new box, and tests fail. Specifically, I get:

```
=$ make test
"/usr/bin/perl" -MExtUtils::Command::MM -e 'cp_nonempty' -- Pg.bs blib/arch/auto/DBD/Pg/Pg.bs 644
PGINITDB="/home/pgdba/work/bin/initdb" PERL_DL_NONLAZY=1 "/usr/bin/perl" "-MExtUtils::Command::MM" "-MTest::Harness" "-e" "undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')" t/*.t
t/00_signature.t ....... skipped: Set the environment variable TEST_SIGNATURE to enable this test
t/00basic.t ............ ok
t/01connect.t .......... 1
t/01connect.t .......... 1/16 #
# DBI                         Version 1.643
# DBD::Pg                     Version 3.16.0
# Perl                        Version 5.36.0
# OS                          linux
# PostgreSQL (compiled)       150001
# PostgreSQL (target)         150001
# PostgreSQL (reported)       PostgreSQL 15.1 on x86_64-pc-linux-gnu, compiled by gcc (Debian 12.2.0-9) 12.2.0, 64-bit
# Default port                5430
# DBI_DSN                     dbi:Pg:
# DBI_USER                    postgres
# Test schema                 dbd_pg_testschema
# PGPORT                      5430
# LANG                        en_US.UTF-8
# Adjusted:                   DBI_DSN
# Looks like you planned 16 tests but ran 15.
t/01connect.t .......... Dubious, test returned 255 (wstat 65280, 0xff00)
Failed 1/16 subtests
        (less 7 skipped subtests: 8 okay)
t/01constants.t ........ ok
t/02attribs.t .......... ok
t/03dbmethod.t ......... ok
t/03smethod.t .......... ok
t/04misc.t ............. ok
t/06bytea.t ............ ok
t/07copy.t ............. ok
t/08async.t ............ ok
t/09arrays.t ........... ok
t/10_pg_error_field.t .. ok
t/12placeholders.t ..... ok
t/20savepoints.t ....... ok
t/30unicode.t .......... ok
t/99cleanup.t .......... ok

Test Summary Report
-------------------
t/01connect.t        (Wstat: 65280 (exited 255) Tests: 15 Failed: 0)
  Non-zero exit status: 255
  Parse errors: Bad plan.  You planned 16 tests but ran 15.
Files=17, Tests=3018, 22 wallclock secs ( 0.14 usr  0.00 sys +  1.70 cusr  0.19 csys =  2.03 CPU)
Result: FAIL
Failed 1/17 test programs. 0/3018 subtests failed.
make: *** [Makefile:964: test_dynamic] Error 255
```

It is possible that the problem is because of non-standard port that my pg is on, but I even provided it with PGPORT...?

Is there any more information that I can provide to help diagnose the problem?

**Comment by depesz** [comment]
For completeness:

```
=$ lsb_release -a
No LSB modules are available.
Distributor ID: Debian
Description:    Debian GNU/Linux bookworm/sid
Release:        n/a
Codename:       bookworm
```

and perl version:

```
=$ perl -V
Summary of my perl5 (revision 5 version 36 subversion 0) configuration:
   
  Platform:
    osname=linux
    osvers=4.19.0
    archname=x86_64-linux-gnu-thread-multi
    uname='linux localhost 4.19.0 #1 smp debian 4.19.0 x86_64 gnulinux '
    config_args='-Dmksymlinks -Dusethreads -Duselargefiles -Dcc=x86_64-linux-gnu-gcc -Dcpp=x86_64-linux-gnu-cpp -Dld=x86_64-linux-gnu-gcc -Dccflags=-DDEBIAN -Wdate-time -D_FORTIFY_SOURCE=2 -g -O2 -ffile-prefix-map=/dummy/build/dir=. -fstack-protector-strong -Wformat -Werror=format-security -Dldflags= -Wl,-z,relro -Dlddlflags=-shared -Wl,-z,relro -Dcccdlflags=-fPIC -Darchname=x86_64-linux-gnu -Dprefix=/usr -Dprivlib=/usr/share/perl/5.36 -Darchlib=/usr/lib/x86_64-linux-gnu/perl/5.36 -Dvendorprefix=/usr -Dvendorlib=/usr/share/perl5 -Dvendorarch=/usr/lib/x86_64-linux-gnu/perl5/5.36 -Dsiteprefix=/usr/local -Dsitelib=/usr/local/share/perl/5.36.0 -Dsitearch=/usr/local/lib/x86_64-linux-gnu/perl/5.36.0 -Dman1dir=/usr/share/man/man1 -Dman3dir=/usr/share/man/man3 -Dsiteman1dir=/usr/local/man/man1 -Dsiteman3dir=/usr/local/man/man3 -Duse64bitint -Dman1ext=1 -Dman3ext=3perl -Dpager=/usr/bin/sensible-pager -Uafs -Ud_csh -Ud_ualarm -Uusesfio -Uusenm -Ui_libutil -Ui_xlocale -Uversiononly -DDEBUGGING=-g -Doptimize=-O2 -dEs -Duseshrplib -Dlibperl=libperl.so.5.36.0'
    hint=recommended
    useposix=true
    d_sigaction=define
    useithreads=define
    usemultiplicity=define
    use64bitint=define
    use64bitall=define
    uselongdouble=undef
    usemymalloc=n
    default_inc_excludes_dot=define
  Compiler:
    cc='x86_64-linux-gnu-gcc'
    ccflags ='-D_REENTRANT -D_GNU_SOURCE -DDEBIAN -fwrapv -fno-strict-aliasing -pipe -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64'
    optimize='-O2 -g'
    cppflags='-D_REENTRANT -D_GNU_SOURCE -DDEBIAN -fwrapv -fno-strict-aliasing -pipe -I/usr/local/include'
    ccversion=''
    gccversion='12.2.0'
    gccosandvers=''
    intsize=4
    longsize=8
    ptrsize=8
    doublesize=8
    byteorder=12345678
    doublekind=3
    d_longlong=define
    longlongsize=8
    d_longdbl=define
    longdblsize=16
    longdblkind=3
    ivtype='long'
    ivsize=8
    nvtype='double'
    nvsize=8
    Off_t='off_t'
    lseeksize=8
    alignbytes=8
    prototype=define
  Linker and Libraries:
    ld='x86_64-linux-gnu-gcc'
    ldflags =' -fstack-protector-strong -L/usr/local/lib'
    libpth=/usr/local/lib /usr/lib/x86_64-linux-gnu /usr/lib /lib/x86_64-linux-gnu /lib
    libs=-lgdbm -lgdbm_compat -ldb -ldl -lm -lpthread -lc -lcrypt
    perllibs=-ldl -lm -lpthread -lc -lcrypt
    libc=/lib/x86_64-linux-gnu/libc.so.6
    so=so
    useshrplib=true
    libperl=libperl.so.5.36
    gnulibc_version='2.35'
  Dynamic Linking:
    dlsrc=dl_dlopen.xs
    dlext=so
    d_dlsymun=undef
    ccdlflags='-Wl,-E'
    cccdlflags='-fPIC'
    lddlflags='-shared -L/usr/local/lib -fstack-protector-strong'


Characteristics of this binary (from libperl): 
  Compile-time options:
    HAS_TIMES
    MULTIPLICITY
    PERLIO_LAYERS
    PERL_COPY_ON_WRITE
    PERL_DONT_CREATE_GVSV
    PERL_MALLOC_WRAP
    PERL_OP_PARENT
    PERL_PRESERVE_IVUV
    USE_64_BIT_ALL
    USE_64_BIT_INT
    USE_ITHREADS
    USE_LARGE_FILES
    USE_LOCALE
    USE_LOCALE_COLLATE
    USE_LOCALE_CTYPE
    USE_LOCALE_NUMERIC
    USE_LOCALE_TIME
    USE_PERLIO
    USE_PERL_ATOF
    USE_REENTRANT_API
    USE_THREAD_SAFE_LOCALE
  Locally applied patches:
    DEBPKG:debian/cpan_definstalldirs - Provide a sensible INSTALLDIRS default for modules installed from CPAN.
    DEBPKG:debian/db_file_ver - https://bugs.debian.org/340047 Remove overly restrictive DB_File version check.
    DEBPKG:debian/doc_info - Replace generic man(1) instructions with Debian-specific information.
    DEBPKG:debian/enc2xs_inc - https://bugs.debian.org/290336 Tweak enc2xs to follow symlinks and ignore missing @INC directories.
    DEBPKG:debian/errno_ver - https://bugs.debian.org/343351 Remove Errno version check due to upgrade problems with long-running processes.
    DEBPKG:debian/libperl_embed_doc - https://bugs.debian.org/186778 Note that libperl-dev package is required for embedded linking
    DEBPKG:fixes/respect_umask - Respect umask during installation
    DEBPKG:debian/writable_site_dirs - Set umask approproately for site install directories
    DEBPKG:debian/extutils_set_libperl_path - EU:MM: set location of libperl.a under /usr/lib
    DEBPKG:debian/no_packlist_perllocal - Don't install .packlist or perllocal.pod for perl or vendor
    DEBPKG:debian/fakeroot - Postpone LD_LIBRARY_PATH evaluation to the binary targets.
    DEBPKG:debian/instmodsh_doc - Debian policy doesn't install .packlist files for core or vendor.
    DEBPKG:debian/ld_run_path - Remove standard libs from LD_RUN_PATH as per Debian policy.
    DEBPKG:debian/libnet_config_path - Set location of libnet.cfg to /etc/perl/Net as /usr may not be writable.
    DEBPKG:debian/perlivp - https://bugs.debian.org/510895 Make perlivp skip include directories in /usr/local
    DEBPKG:debian/squelch-locale-warnings - https://bugs.debian.org/508764 Squelch locale warnings in Debian package maintainer scripts
    DEBPKG:debian/patchlevel - https://bugs.debian.org/567489 List packaged patches for 5.36.0-4 in patchlevel.h
    DEBPKG:fixes/document_makemaker_ccflags - https://bugs.debian.org/628522 [rt.cpan.org #68613] Document that CCFLAGS should include $Config{ccflags}
    DEBPKG:debian/find_html2text - https://bugs.debian.org/640479 Configure CPAN::Distribution with correct name of html2text
    DEBPKG:debian/perl5db-x-terminal-emulator.patch - https://bugs.debian.org/668490 Invoke x-terminal-emulator rather than xterm in perl5db.pl
    DEBPKG:debian/cpan-missing-site-dirs - https://bugs.debian.org/688842 Fix CPAN::FirstTime defaults with nonexisting site dirs if a parent is writable
    DEBPKG:fixes/memoize_storable_nstore - [rt.cpan.org #77790] https://bugs.debian.org/587650 Memoize::Storable: respect 'nstore' option not respected
    DEBPKG:debian/makemaker-pasthru - https://bugs.debian.org/758471 Pass LD settings through to subdirectories
    DEBPKG:debian/makemaker-manext - https://bugs.debian.org/247370 Make EU::MakeMaker honour MANnEXT settings in generated manpage headers
    DEBPKG:debian/kfreebsd-softupdates - https://bugs.debian.org/796798 Work around Debian Bug#796798
    DEBPKG:fixes/memoize-pod - [rt.cpan.org #89441] Fix POD errors in Memoize
    DEBPKG:debian/hurd-softupdates - https://bugs.debian.org/822735 Fix t/op/stat.t failures on hurd
    DEBPKG:fixes/math_complex_doc_great_circle - https://bugs.debian.org/697567 [rt.cpan.org #114104] Math::Trig: clarify definition of great_circle_midpoint
    DEBPKG:fixes/math_complex_doc_see_also - https://bugs.debian.org/697568 [rt.cpan.org #114105] Math::Trig: add missing SEE ALSO
    DEBPKG:fixes/math_complex_doc_angle_units - https://bugs.debian.org/731505 [rt.cpan.org #114106] Math::Trig: document angle units
    DEBPKG:fixes/cpan_web_link - https://bugs.debian.org/367291 CPAN: Add link to main CPAN web site
    DEBPKG:debian/hppa_op_optimize_workaround - https://bugs.debian.org/838613 Temporarily lower the optimization of op.c on hppa due to gcc-6 problems
    DEBPKG:debian/installman-utf8 - https://bugs.debian.org/840211 Generate man pages with UTF-8 characters
    DEBPKG:debian/hppa_opmini_optimize_workaround - https://bugs.debian.org/869122 Lower the optimization level of opmini.c on hppa
    DEBPKG:debian/sh4_op_optimize_workaround - https://bugs.debian.org/869373 Also lower the optimization level of op.c and opmini.c on sh4
    DEBPKG:debian/perldoc-pager - https://bugs.debian.org/870340 [rt.cpan.org #120229] Fix perldoc terminal escapes when sensible-pager is less
    DEBPKG:debian/prune_libs - https://bugs.debian.org/128355 Prune the list of libraries wanted to what we actually need.
    DEBPKG:debian/mod_paths - Tweak @INC ordering for Debian
    DEBPKG:debian/deprecate-with-apt - https://bugs.debian.org/747628 Point users to Debian packages of deprecated core modules
    DEBPKG:debian/disable-stack-check - https://bugs.debian.org/902779 [GH #16607] Disable debugperl stack extension checks for binary compatibility with perl
    DEBPKG:debian/perlbug-editor - https://bugs.debian.org/922609 Use "editor" as the default perlbug editor, as per Debian policy
    DEBPKG:debian/eu-mm-perl-base - https://bugs.debian.org/962138 Suppress an ExtUtils::MakeMaker warning about our non-default @INC
    DEBPKG:fixes/io_socket_ip_ipv6 - Disable getaddrinfo(3) AI_ADDRCONFIG for localhost and IPv4 numeric addresses
    DEBPKG:debian/usrmerge-lib64 - https://bugs.debian.org/914128 Configure / libpth.U: Do not adjust glibpth when /usr/lib64 is present.
    DEBPKG:debian/usrmerge-realpath - https://bugs.debian.org/914128 Configure / libpth.U: use realpath --no-symlinks on Debian
    DEBPKG:debian/configure-regen - https://bugs.debian.org/762638 Regenerate Configure et al. after probe unit changes
    DEBPKG:fixes/x32-io-msg-skip - https://bugs.debian.org/922609 Skip io/msg.t on x32 due to broken System V message queues
    DEBPKG:debian/hurd-eumm-workaround - https://bugs.debian.org/1018289 Work around a MakeMaker regression breaking GNU/Hurd hint files
    DEBPKG:fixes/json-pp-warnings - https://bugs.debian.org/1019757 Call unimport first to silence warnings
  Built under linux
  Compiled at Oct 18 2022 16:44:58
  %ENV:
    PERL5LIB="/home/depesz/perl5/lib/perl5:/home/depesz/perl5/lib/perl5:/home/depesz/perl5/lib/perl5"
    PERL_LOCAL_LIB_ROOT="/home/depesz/perl5:/home/depesz/perl5:/home/depesz/perl5"
    PERL_MB_OPT="--install_base "/home/depesz/perl5""
    PERL_MM_OPT="INSTALL_BASE=/home/depesz/perl5"
  @INC:
    /home/depesz/perl5/lib/perl5/5.36.0/x86_64-linux-gnu-thread-multi
    /home/depesz/perl5/lib/perl5/5.36.0
    /home/depesz/perl5/lib/perl5/x86_64-linux-gnu-thread-multi
    /home/depesz/perl5/lib/perl5
    /home/depesz/perl5/lib/perl5/5.36.0/x86_64-linux-gnu-thread-multi
    /home/depesz/perl5/lib/perl5/5.36.0
    /home/depesz/perl5/lib/perl5/x86_64-linux-gnu-thread-multi
    /home/depesz/perl5/lib/perl5
    /home/depesz/perl5/lib/perl5/5.36.0/x86_64-linux-gnu-thread-multi
    /home/depesz/perl5/lib/perl5/5.36.0
    /home/depesz/perl5/lib/perl5/x86_64-linux-gnu-thread-multi
    /home/depesz/perl5/lib/perl5
    /etc/perl
    /usr/local/lib/x86_64-linux-gnu/perl/5.36.0
    /usr/local/share/perl/5.36.0
    /usr/lib/x86_64-linux-gnu/perl5/5.36
    /usr/share/perl5
    /usr/lib/x86_64-linux-gnu/perl-base
    /usr/lib/x86_64-linux-gnu/perl/5.36
    /usr/share/perl/5.36
    /usr/local/lib/site_perl

```

**Comment by ilmari** [comment]
This looks like the test skip count that got fixed in 37480c50bf1a34a7d81b99c52de68ca3a9483795 (and will be in 3.16.1).

To verify, could you run this command and see how many `DBI_DSN contains no database option, so skipping connection tests` `SKIP` lines it outputs?

    make test TEST_VERBOSE=1 TEST_FILES='t/01connect.t t/99cleanup.t'

**Comment by depesz** [comment]
I think that none:

```
=$ make test TEST_VERBOSE=1 TEST_FILES='t/01connect.t t/99cleanup.t'
"/usr/bin/perl" -MExtUtils::Command::MM -e 'cp_nonempty' -- Pg.bs blib/arch/auto/DBD/Pg/Pg.bs 644
PGINITDB="/home/pgdba/work/bin/initdb" PERL_DL_NONLAZY=1 "/usr/bin/perl" "-MExtUtils::Command::MM" "-MTest::Harness" "-e" "undef *Test::Harness::Switches; test_harness(1, 'blib/lib', 'blib/arch')" t/01connect.t t/99cleanup.t
t/01connect.t .. Please wait, creating new database (version 15.1) for testing
netstat call failed, trying port 5442

1..16
ok 1 - Established a connection to the database
ok 2 - Disconnect from the database
ok 3 - Second database connection attempt worked
ok 4 - Connected with second database handle
ok 5 - Disconnect with first database handle
ok 6 - Disconnect with second database handle
ok 7 - Disconnect again with second database handle
ok 8 - Execute fails on a disconnected statement
ok 9 - Connect with invalid option fails
ok 10 - Connect using string 'db' works
ok 11 - Connect using string 'dbname' works
ok 12 - Connect using string 'database' works
ok 13 - Connect with forced uppercase 'DBI:' works
ok 14 - Connect with mixed case 'DbI:' works
ok 15 - Connect with multiple host names works
ok 16 - Connect using a quoted database argument
#
# DBI                         Version 1.643
# DBD::Pg                     Version 3.16.0
# Perl                        Version 5.36.0
# OS                          linux
# PostgreSQL (compiled)       150001
# PostgreSQL (target)         150001
# PostgreSQL (reported)       PostgreSQL 15.1 on x86_64-pc-linux-gnu, compiled by gcc (Debian 12.2.0-9) 12.2.0, 64-bit
# Default port                5430
# DBI_DSN                     dbi:Pg:db="postgres";port=5442;host=<pwd>/dbdpg_test_database/data/socket
# DBI_USER                    depesz
# Test schema                 dbd_pg_testschema
# LANG                        C
# Adjusted:                   initdb
ok
t/99cleanup.t ..
1..1
ok 1 - Connect to database for cleanup
ok
All tests successful.
Files=2, Tests=17,  2 wallclock secs ( 0.01 usr  0.00 sys +  0.35 cusr  0.19 csys =  0.55 CPU)
Result: PASS
```

**Comment by portnov** [comment]
Is there any workaround to install DBD::pg from CPAN?

**Comment by esabol** [comment]
> Is there any workaround to install DBD::pg from CPAN?

Force install it. https://www.thegeekstuff.com/2013/06/cpan-force-install-perl-module/

If you are using `cpanm`, add the `-f` switch.

**Comment by keepflying1** [comment]
apologies for the delay @esabol - holidays and a wedding!

to be perfectly honest, I don't know how to apply the changes to a test suite. could you ELI5?

**Comment by esabol** [comment]
OK....

1. Open this link for reference: https://github.com/bucardo/dbdpg/commit/37480c50bf1a34a7d81b99c52de68ca3a9483795
2. Load the file `t/01connect.t` into your favorite text editor.
3. Make the change show in the link from step 1.
4. Load the file `t/dbdpg_test_setup.pl` into your favorite text editor.
5. Make the change show in the link from step 1.
6. Then type `make test`.

**Comment by ilmari** [comment]
This was fixed in 3.16.1  (released March 5, 2023).
